File: qt/gen_qmessageauthenticationcode.go
package qt
import (
	"unsafe"
)

		type QMessageAuthenticationCode struct {
			h uintptr
			isSubclass bool
}
		
			// NewQMessageAuthenticationCode constructs a new QMessageAuthenticationCode object.
			func NewQMessageAuthenticationCode(method QCryptographicHash__Algorithm) *QMessageAuthenticationCode {
								
				ret := newQMessageAuthenticationCode(QMessageAuthenticationCode_new((int)(method)))
				ret.isSubclass = true
				return ret
			}
			
			
			// NewQMessageAuthenticationCode2 constructs a new QMessageAuthenticationCode object.
			func NewQMessageAuthenticationCode2(method QCryptographicHash__Algorithm, key QByteArrayView) *QMessageAuthenticationCode {
								
				ret := newQMessageAuthenticationCode(QMessageAuthenticationCode_new2((int)(method), key.cPointer()))
				ret.isSubclass = true
				return ret
			}
			
			
			func (this *QMessageAuthenticationCode) Swap(other *QMessageAuthenticationCode)  {
				 QMessageAuthenticationCode_Swap(this.h, other.cPointer())
}
			
			func (this *QMessageAuthenticationCode) Reset()  {
				 QMessageAuthenticationCode_Reset(this.h)
}
			
			func (this *QMessageAuthenticationCode) SetKey(key QByteArrayView)  {
				 QMessageAuthenticationCode_SetKey(this.h, key.cPointer())
}
			
			func (this *QMessageAuthenticationCode) AddData(data string, length int64)  {
				data_Cstring := CString(data)
defer free(unsafe.Pointer(data_Cstring))
 QMessageAuthenticationCode_AddData(this.h, data_Cstring, (ptrdiff_t)(length))
}
			
			func (this *QMessageAuthenticationCode) AddDataWithData(data QByteArrayView)  {
				 QMessageAuthenticationCode_AddDataWithData(this.h, data.cPointer())
}
			
			func (this *QMessageAuthenticationCode) AddDataWithDevice(device *QIODevice) bool {
				return (bool)(QMessageAuthenticationCode_AddDataWithDevice(this.h, device.cPointer()))
}
			
			func (this *QMessageAuthenticationCode) ResultView() *QByteArrayView {
				_goptr := newQByteArrayView(QMessageAuthenticationCode_ResultView(this.h))
_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
return _goptr
}
			
			func (this *QMessageAuthenticationCode) Result() []byte {
				var _bytearray struct_miqt_string =  QMessageAuthenticationCode_Result(this.h)
_ret := GoBytes(unsafe.Pointer(_bytearray.data), int(int64(_bytearray.len)))
free(unsafe.Pointer(_bytearray.data))
return _ret}
			
			func QMessageAuthenticationCode_Hash(message QByteArrayView, key QByteArrayView, method QCryptographicHash__Algorithm) []byte {
				var _bytearray struct_miqt_string =  QMessageAuthenticationCode_Hash(message.cPointer(), key.cPointer(), (int)(method))
_ret := GoBytes(unsafe.Pointer(_bytearray.data), int(int64(_bytearray.len)))
free(unsafe.Pointer(_bytearray.data))
return _ret}
			
			func QMessageAuthenticationCode_HashInto(buffer QSpan<char>, message QByteArrayView, key QByteArrayView, method QCryptographicHash__Algorithm) *QByteArrayView {
				_goptr := newQByteArrayView(QMessageAuthenticationCode_HashInto(buffer, message.cPointer(), key.cPointer(), (int)(method)))
_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
return _goptr
}
			
			func QMessageAuthenticationCode_HashInto2(buffer QSpan<uchar>, message QByteArrayView, key QByteArrayView, method QCryptographicHash__Algorithm) *QByteArrayView {
				_goptr := newQByteArrayView(QMessageAuthenticationCode_HashInto2(buffer, message.cPointer(), key.cPointer(), (int)(method)))
_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
return _goptr
}
			
			func QMessageAuthenticationCode_HashInto3(buffer QSpan<std__byte>, message QByteArrayView, key QByteArrayView, method QCryptographicHash__Algorithm) *QByteArrayView {
				_goptr := newQByteArrayView(QMessageAuthenticationCode_HashInto3(buffer, message.cPointer(), key.cPointer(), (int)(method)))
_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
return _goptr
}
			
			func QMessageAuthenticationCode_HashInto4(buffer QSpan<char>, messageParts QSpan<const QByteArrayView>, key QByteArrayView, method QCryptographicHash__Algorithm) *QByteArrayView {
				_goptr := newQByteArrayView(QMessageAuthenticationCode_HashInto4(buffer, messageParts, key.cPointer(), (int)(method)))
_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
return _goptr
}
			
			func QMessageAuthenticationCode_HashInto5(buffer QSpan<uchar>, messageParts QSpan<const QByteArrayView>, key QByteArrayView, method QCryptographicHash__Algorithm) *QByteArrayView {
				_goptr := newQByteArrayView(QMessageAuthenticationCode_HashInto5(buffer, messageParts, key.cPointer(), (int)(method)))
_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
return _goptr
}
			
			func QMessageAuthenticationCode_HashInto6(buffer QSpan<std__byte>, message QSpan<const QByteArrayView>, key QByteArrayView, method QCryptographicHash__Algorithm) *QByteArrayView {
				_goptr := newQByteArrayView(QMessageAuthenticationCode_HashInto6(buffer, message, key.cPointer(), (int)(method)))
_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
return _goptr
}
			

File: qt/gen_qmessageauthenticationcode.cpp
// +build ignore

#include <QByteArray>
#include <QByteArrayView>
#include <QIODevice>
#include <QMessageAuthenticationCode>
#include <qmessageauthenticationcode.h>
#include "gen_qmessageauthenticationcode.h"

#ifndef _Bool
#define _Bool bool
#endif

QMessageAuthenticationCode* QMessageAuthenticationCode_new(int method) {
	return new QMessageAuthenticationCode(static_cast<QCryptographicHash::Algorithm>(method));
}

QMessageAuthenticationCode* QMessageAuthenticationCode_new2(int method, QByteArrayView* key) {
	return new QMessageAuthenticationCode(static_cast<QCryptographicHash::Algorithm>(method), *key);
}

void QMessageAuthenticationCode_Swap(QMessageAuthenticationCode* self, QMessageAuthenticationCode* other) {
	self->swap(*other);
}

void QMessageAuthenticationCode_Reset(QMessageAuthenticationCode* self) {
	self->reset();
}

void QMessageAuthenticationCode_SetKey(QMessageAuthenticationCode* self, QByteArrayView* key) {
	self->setKey(*key);
}

void QMessageAuthenticationCode_AddData(QMessageAuthenticationCode* self, const char* data, ptrdiff_t length) {
	self->addData(data, (qsizetype)(length));
}

void QMessageAuthenticationCode_AddDataWithData(QMessageAuthenticationCode* self, QByteArrayView* data) {
	self->addData(*data);
}

bool QMessageAuthenticationCode_AddDataWithDevice(QMessageAuthenticationCode* self, QIODevice* device) {
	return self->addData(device);
}

QByteArrayView* QMessageAuthenticationCode_ResultView(const QMessageAuthenticationCode* self) {
	return new QByteArrayView(self->resultView());
}

struct miqt_string QMessageAuthenticationCode_Result(const QMessageAuthenticationCode* self) {
	QByteArray _qb = self->result();
	struct miqt_string _ms;
	_ms.len = _qb.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _qb.data(), _ms.len);
	return _ms;
}

struct miqt_string QMessageAuthenticationCode_Hash(QByteArrayView* message, QByteArrayView* key, int method) {
	QByteArray _qb = QMessageAuthenticationCode::hash(*message, *key, static_cast<QCryptographicHash::Algorithm>(method));
	struct miqt_string _ms;
	_ms.len = _qb.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _qb.data(), _ms.len);
	return _ms;
}

QByteArrayView* QMessageAuthenticationCode_HashInto(QSpan<char> buffer, QByteArrayView* message, QByteArrayView* key, int method) {
	return new QByteArrayView(QMessageAuthenticationCode::hashInto(buffer, *message, *key, static_cast<QCryptographicHash::Algorithm>(method)));
}

QByteArrayView* QMessageAuthenticationCode_HashInto2(QSpan<uchar> buffer, QByteArrayView* message, QByteArrayView* key, int method) {
	return new QByteArrayView(QMessageAuthenticationCode::hashInto(buffer, *message, *key, static_cast<QCryptographicHash::Algorithm>(method)));
}

QByteArrayView* QMessageAuthenticationCode_HashInto3(QSpan<std::byte> buffer, QByteArrayView* message, QByteArrayView* key, int method) {
	return new QByteArrayView(QMessageAuthenticationCode::hashInto(buffer, *message, *key, static_cast<QCryptographicHash::Algorithm>(method)));
}

QByteArrayView* QMessageAuthenticationCode_HashInto4(QSpan<char> buffer, QSpan<const QByteArrayView> messageParts, QByteArrayView* key, int method) {
	return new QByteArrayView(QMessageAuthenticationCode::hashInto(buffer, messageParts, *key, static_cast<QCryptographicHash::Algorithm>(method)));
}

QByteArrayView* QMessageAuthenticationCode_HashInto5(QSpan<uchar> buffer, QSpan<const QByteArrayView> messageParts, QByteArrayView* key, int method) {
	return new QByteArrayView(QMessageAuthenticationCode::hashInto(buffer, messageParts, *key, static_cast<QCryptographicHash::Algorithm>(method)));
}

QByteArrayView* QMessageAuthenticationCode_HashInto6(QSpan<std::byte> buffer, QSpan<const QByteArrayView> message, QByteArrayView* key, int method) {
	return new QByteArrayView(QMessageAuthenticationCode::hashInto(buffer, message, *key, static_cast<QCryptographicHash::Algorithm>(method)));
}

void QMessageAuthenticationCode_Delete(QMessageAuthenticationCode* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QMessageAuthenticationCode*>( self );
	} else {
		delete self;
	}
}

